lib/cstring.h

```diff
@@ -356,6 +358,16 @@ inline std::ostream &operator<<(std::ostream &out, cstring s) {
     return out << (s ? s.c_str() : "<null>");
 }
 
+/// Let's prevent literal clashes. A user wishing to use the literal can do using namespace
+/// P4::literals, similarly as they can do using namespace std::literals for the standard once.
+namespace P4::literals {
+
+/// A user-provided literal suffix to allow creation of cstring from literals: "foo"_cs.
+/// Note that the C++ standard mandates that all user-defined literal suffixes defined outside of
+/// the standard library must start with underscore.
+inline cstring operator""_cs(const char *str, std::size_t len) { return cstring(str, len); }
```

Comment:
By the way, why it essentially makes copy of a literal? Shouldn't this be semantically equivalent to `cstring::literal`?

Reply:
I just didn't notice `cstring::literal`, feel free to change the `_cs`.

Reply:
Ok, great! So, there is no particular reason that I'm missing here :)